src/Altinn.App.Core/Internal/Process/ProcessEngine.cs

```diff
@@ -131,7 +131,7 @@ public async Task<ProcessChangeResult> Next(ProcessNextRequest request)
                 ErrorType = ProcessErrorType.Conflict
             };
         }
-        var actionHandler = await _userActionFactory.GetActionHandlerOrDefault(request.Action, new NullUserAction()).HandleAction(new UserActionContext(request.Instance, userId.Value));
+        var actionHandler = await _userActionService.GetActionHandlerOrDefault(request.Action, new NullUserAction()).HandleAction(new UserActionContext(request.Instance, userId.Value));
```

Comment:
Is it actually correct to have an action with no implementation, so that the `NullUserAction` is required?

Reply:
You can use the action Id to navigate the process without having any custom logic connected to it. In those cases I think we shouldn't force the developer to implement a empty IUserAction and just default to the `NullUserAction`

Reply:
In that case I'd rather check the null return value from `UserActionServcie.GetActionHandler`, so the code explicitly states how to handle a case without custom logic.